modules/carto/src/api/layer-map.ts

```diff
@@ -380,6 +372,33 @@ export function getColorAccessor(
   return normalizeAccessor(accessor, data);
 }
 
+export function getDomainAndScaleColor(name, scaleType, range, data) {
```

Comment:
Why do you need to introduce this new function? I feel it just makes the code harder to follow

Reply:
I don't need but I think that it's more clear than having a long function. This function is just calculating the `domain` and `scaleColor` for each case.

Reply:
OK, but can your rename to `calculateLayerScale`, remove the `export` and put these three lines in the helper function:

```
scale.domain(domain);
scale.range(scaleColor);
scale.unknown(UNKNOWN_COLOR);
```

and then just return the `scale`